### DISCUSSION TITLE
Multiple Conda env exceed Github Act. OS disk space

### DISCUSSION THREAD

**Discussion by stefanbringuier** [rodluger/showyourwork#418] [Q&A]
In relation to #414, I have multiple `Conda` environments, as a result the `./snakemake/conda` folder is massive, ~36 GB. This is causing the Github `build` action to fail because the runners only have 25-29 GB. I tried to use the https://github.com/marketplace/actions/maximize-build-disk-space Github action, but still got the same issue:

```shell
NoSpaceLeftError: No space left on devices.
```

I'm trying to think about ways to:

1. reduce the size of the `./snakemake/conda` folder via `showyourwork build --conda-frontend=mamba`. Not sure how to invoke this in the `showyourwork` Github action.
2. Push/checkout locally `main` to `main-pdf`, run `showyourwork build` and tar locally, then push to Github.

I think the 2. option is a reasonable option for my case, but not really sure how best to do this without making this ugly and also deviates from the ethos of `showyourwork`.

**Comment by HealthyPear** [comment]
Not sure if this is still relevant for you, but I think a nice idea would be to help out with

https://github.com/showyourwork/showyourwork-action/issues/11

so anybody could use any public server not hosted on github

**Reply by stefanbringuier** [reply]
Thank you for the callout. The issue is that I have moved on to `uv` now. I have also found that `showyourwork` is less suited for simulation+analysis, but rather just for the analysis reproducibility part.

**Reply by HealthyPear** [reply]
> The issue is that I have moved on to `uv` now.

Not sure how much we can go into the direction of supporting uv in the near future, as being based on Snakemake we are bound to conda even though there is people asking for this (see e.g. https://github.com/snakemake/snakemake/issues/3251).

> I have also found that `showyourwork` is less suited for simulation+analysis, but rather just for the analysis reproducibility part.

Could you elaborate on this a bit more? I am interested in user feedback. Is it mainly because of the storage size of the simulation files or there is some feature that you think it's missing?
